processing/src/main/java/io/druid/segment/incremental/OffheapIncrementalIndex.java

```diff
@@ -199,6 +192,15 @@ protected Integer addToFacts(
       if (prev != null) {
         rowOffset = prev;
       } else {
+        rowContainer.set(row);
+        for (int i = 0; i < metrics.length; i++) {
+          final AggregatorFactory agg = metrics[i];
+          aggs[i] = agg.factorizeBuffered(
+              makeColumnSelectorFactory(agg, rowSupplier, deserializeComplexMetrics)
+          );
+        }
+        rowContainer.set(null);
```

Comment:
We may make one set/unset to cover all the code of addToFacts.
In my thought, however, it is better to keep two sets of set/unset 
because there exist several error checking and exception throwing routines between those two and with one big set/unset, we should add unset before all the exception throwing that may be catched.